keyboards/kprepublic/bm60rgb/rev2/keymaps/default/keymap.c

```diff
@@ -0,0 +1,91 @@
+/* Copyright 2021 bdtc123
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+enum my_keycodes {
+    RMT = SAFE_RANGE,
+    RMS,
+    RMIH,
+    RMDH,
+    RMIS,
+    RMDS,
+    RMIV,
+    RMDV
+};
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+  switch (keycode) {
+    case RMT:
+      if (record->event.pressed)
+      {rgb_matrix_toggle();
+      }
+      return false;
+    case RMS:
+      if (record->event.pressed)
+      {rgb_matrix_step();
+      }
+      return false;
+      case RMIH:
+      if (record->event.pressed)
+      {rgb_matrix_increase_hue();
+      }
+      return false;
+      case RMDH:
+      if (record->event.pressed)
+      {rgb_matrix_decrease_hue();
+      }
+      return false;
+      case RMIS:
+      if (record->event.pressed)
+      {rgb_matrix_increase_sat();
+      }
+      return false;
+      case RMDS:
+      if (record->event.pressed)
+      {rgb_matrix_decrease_sat();
+      }
+      return false;
+      case RMIV:
+      if (record->event.pressed)
+      {rgb_matrix_increase_val();
+      }
+      return false;
+      case RMDV:
+      if (record->event.pressed)
+      {rgb_matrix_decrease_val();
+      }
+      return true;
+      default:
+      return true;
+  }
+}
```

Comment:
```suggestion
#define RMT  RGB_TOG
#define RMS  RGB_MOD
#define RMIH RGB_HUI
#define RMDH RGB_HUD
#define RMIS RGB_SAI
#define RMDS RGB_SAD
#define RMIV RGB_VAI
#define RMDV RGB_VAD
```
You can reduce code size by using built in key codes